apkid/rules/dex/obfuscators.yara

```diff
@@ -241,3 +241,19 @@ rule aamo_str_enc : obfuscator
   condition:
     1 of ($opcodes*) and all of ($a, $b)
 }
+
+rule kiwi_amazon : obfuscator
+{
+    meta:
+        description = "Kiwi encrypter"
+        sample      = "3e309548f90160e3a4dc6f67621c75d2b66cc3b580da7306ff3dc6d6c25bb8a1"
+        author      = "Eduardo Novella"
+
+    strings:
+        $key   = "Kiwi__Version__Obfuscator"
```

Comment:
Can these be dex strings?

Reply:
Umm nope?

```c
0000h: 64 65 78 0A 30 33 35 00 D5 52 4A A8 97 2F C7 1E  dex.035.ÕRJ¨—/Ç.
0010h: AE CD 43 E4 34 E2 EF BF 55 ED FB FE 85 DE 83 D5  ®ÍCä4âï¿Uíûþ…ÞƒÕ
0020h: 80 C2 7A 00 70 00 00 00 78 56 34 12 00 00 00 00  €Âz.p...xV4.....
0030h: 00 00 00 00 A4 C1 7A 00 C7 E5 00 00 70 00 00 00  ....¤Áz.Çå..p...
0040h: A1 29 00 00 8C 97 03 00 06 37 00 00 4B 69 77 69  ¡)..Œ—...7..Kiwi
0050h: 5F 5F 56 65 72 73 69 6F 6E 5F 5F 4F 62 66 75 73  __Version__Obfus
0060h: 63 61 74 6F 72 5D 12 00 40 8F 64 00 40 33 16 00  cator]..@
```

Reply:
You say it can't be done then post the dex string? 🤔

The dex string is `00 4B 69 77 69 5F 5F 56 65 72 73 69 6F 6E 5F 5F 4F 62 66 75 73 63 61 74 6F 72 5D 12 00`

Reply:
We can maybe add more:
```c
[0x00027044]> x @ 0xdc36
- offset -   0 1  2 3  4 5  6 7  8 9  A B  C D  E F  0123456789ABCDEF
0x0000dc36  2f4c 636f 6d2f 616d 617a 6f6e 2f6d 6173  /Lcom/amazon/mas
0x0000dc46  2f6b 6977 692f 7574 696c 2f4b 6977 6956  /kiwi/util/KiwiV
0x0000dc56  6572 7369 6f6e 456e 6372 7970 7465 723b  ersionEncrypter;
0x0000dc66  0028 4c63 6f6d 2f61 6d61 7a6f 6e2f 6d61  .(Lcom/amazon/ma

```

Reply:
I will add these dex string now...
```c
[19:04 edu@de11 kiwi-obfuscator] >  egrep -rin obfuscator
Binary file f91459e26e609fb12637d10719c82345e658be04774ebcc492e26876bbb9e059.unpack/classes3.dex matches
[19:04 edu@de11 kiwi-obfuscator] >  r2 f91459e26e609fb12637d10719c82345e658be04774ebcc492e26876bbb9e059.unpack/classes3.dex
 -- See you in shell
[0x00027044]> izzq~+obfuscator
0xbf4c 26 25 Kiwi__Version__Obfuscator
[0x00027044]> x 32 @ 0xbf4c-4
- offset -   0 1  2 3  4 5  6 7  8 9  A B  C D  E F  0123456789ABCDEF
0x0000bf48  7661 0019 4b69 7769 5f5f 5665 7273 696f  va..Kiwi__Versio
0x0000bf58  6e5f 5f4f 6266 7573 6361 746f 7200 014c  n__Obfuscator..L
[0x00027044]> 
```

Reply:
BINGO!
```c
[19:14 edu@ohyeah kiwi_ob] >  ll
total 120M
-rw-rw-r-- 1 edu edu 12M Nov 24 19:13 5e86b8da19710aab385b48297d59f3cf2eeedb43efdb0b9ccac7fc9f7af93642
-rw-rw-r-- 1 edu edu 34M Nov 24 19:13 6db2e2756b62ec97de704ed323ffc7bdbf5ad49c481da7dff7b9dc6f64afc557
-rw-rw-r-- 1 edu edu 27M Nov 24 19:13 7937de1a75d510baeabd80c64285dfb0c94ea2a1c019f7fb046b388d03555b83
-rw-rw-r-- 1 edu edu 48M Nov 24 19:13 5545823d92efb74ccb96ab53b709940456431bce0ffe894dde8205f59d5972fc
[19:14 edu@ohyeah kiwi_ob] >  apkid . |grep Kiwi
 |-> obfuscator : Kiwi encrypter
 |-> obfuscator : Kiwi encrypter
 |-> obfuscator : Kiwi encrypter
 |-> obfuscator : Kiwi encrypter

```